newIDE/app/src/VariablesList/index.js

```diff
@@ -149,22 +150,42 @@ export default class VariablesList extends React.Component<Props, State> {
           variable,
           depth + 1,
           index,
-          parentVariable
+          parentVariable,
+          type
         );
       })
     );
   }
 
+  _getInstanceVariableType = (name: string) => {
+    const { variablesContainer, inheritedVariablesContainer } = this.props;
+    return inheritedVariablesContainer // We check for 3 types of variable states, when editing instance variables
+      ? variablesContainer.has(name) && !inheritedVariablesContainer.has(name)
+        ? 'instance' // -variable that is unique to the instance - we can edit its name/structure
+        : !variablesContainer.has(name) && inheritedVariablesContainer.has(name)
+        ? 'object' // -displaying object variable at its default value, changing it will create an inherited
+        : variablesContainer.has(name) && inheritedVariablesContainer.has(name)
+        ? 'inherited' // -instance variable created when editing an object variable's value, deleting it will re-show
+        : '' // object go in inheritedVariablesContainer, the rest stay in variablesContainer
+      : '';
+  };
+
   _renderVariableAndChildrenRows(
     name: string,
     variable: gdVariable,
     depth: number,
     index: number,
-    parentVariable: ?gdVariable
+    parentVariable: ?gdVariable,
+    parentType: ?string = ''
   ) {
-    const { variablesContainer } = this.props;
+    const { variablesContainer, inheritedVariablesContainer } = this.props;
     const isStructure = variable.isStructure();
 
+    const type =
+      parentType && parentType.length
+        ? parentType // a state can come from a parent variable
```

Comment:
`// a state can come from a parent variable` is a bit of a too generic comment, and does not provides value compared to the code.

Also avoid passing empty strings for things that are not set, because it actually can be a real value. Pass `null` instead, and check for `null`:
`const origin = parentOrigin !== null ? parentOrigin :  this._getVariableOrigin(name)`